src/store/reducers/keymap.ts

```diff
@@ -62,7 +62,12 @@ export default function(state = initialState, action: Action): KeymapState {
 
         case KeymapActions.SET_DEFAULT:
             newState = state.entities.map((keymap: Keymap) => {
-                keymap.isDefault = (keymap.abbreviation === action.payload);
+                if (keymap.abbreviation === action.payload || keymap.isDefault) {
+                    let newKeymap: Keymap = new Keymap();
+                    Object.assign(newKeymap, keymap);
+                    keymap = newKeymap;
```

Comment:
I wanted to have one return statement., because this way the code more readable for me. 
If you need a new keymap, then you create a new and replace the old one, otherwise it remains the same. I know that your approach does the same, but I think mine is easier to understand.